src/wayland/output/handlers.rs

```diff
@@ -0,0 +1,211 @@
+use slog::{trace, warn};
+use wayland_protocols::unstable::xdg_output::v1::server::{
+    zxdg_output_manager_v1::{self, ZxdgOutputManagerV1},
+    zxdg_output_v1::{self, ZxdgOutputV1},
+};
+use wayland_server::{
+    protocol::wl_output::{self, Mode as WMode, WlOutput},
+    DelegateDispatch, DelegateDispatchBase, DelegateGlobalDispatch, DelegateGlobalDispatchBase,
+    DestructionNotify, Dispatch, GlobalDispatch, Resource,
+};
+
+use super::{xdg::XdgOutput, Output, OutputGlobalData, OutputManagerState, OutputUserData};
+
+/*
+ * Wl Output
+ */
+
+impl DelegateGlobalDispatchBase<WlOutput> for OutputManagerState {
+    type GlobalData = OutputGlobalData;
+}
+
+impl<D> DelegateGlobalDispatch<WlOutput, D> for OutputManagerState
+where
+    D: GlobalDispatch<WlOutput, GlobalData = OutputGlobalData>,
+    D: Dispatch<WlOutput, UserData = OutputUserData>,
+    D: 'static,
+{
+    fn bind(
+        _state: &mut D,
+        dh: &mut wayland_server::DisplayHandle<'_>,
+        _client: &wayland_server::Client,
+        resource: wayland_server::New<WlOutput>,
+        global_data: &Self::GlobalData,
+        data_init: &mut wayland_server::DataInit<'_, D>,
+    ) {
+        let output = data_init.init(
+            resource,
+            OutputUserData {
+                global_data: global_data.clone(),
+            },
+        );
+
+        let mut inner = global_data.inner.0.lock().unwrap();
+
+        trace!(inner.log, "New global instantiated.");
```

Comment:
If we're going to have these trace messages, maybe they could be a little more informative, maybe at least tell that it is an output that is instantiated.

Reply:
Indeed, not it does say output global, and prints name of the output as well